go/multiorch/coordinator/leader_appointment.go

```diff
@@ -195,7 +194,31 @@ func (c *Coordinator) selectCandidate(ctx context.Context, cohort []*Node) (*Nod
 			continue
 		}
 
-		if bestCandidate == nil || status.walPosition > bestWAL {
+		// Validate LSN format first
+		if _, err := lsn.Parse(status.walPosition); err != nil {
```

Comment:
I've been pondering about this. Under which case could we get a response, but a bad format LSN? This seems like an invariant we can expect to be true.

What do you think if we actually return error in if any LSN is invalid? 

It seems this is the kind of error where we better bail and get an operator to understand why we would be getting LSN's in the response that we can't parse